go/vt/vttablet/tabletmanager/rpc_vreplication.go

```diff
@@ -328,6 +538,80 @@ func (tm *TabletManager) UpdateVReplicationWorkflow(ctx context.Context, req *ta
 	}, nil
 }
 
+// UpdateVReplicationWorkflowsState operates in much the same way that
+// UpdateVReplicationWorkflow does, but it allows you to update the state
+// related fields -- state, message, stop_pos -- for multiple workflows.
+// Note: today this is only used during Reshard as all of the vreplication
+// streams need to be migrated from the old shards to the new ones.
+func (tm *TabletManager) UpdateVReplicationWorkflowsState(ctx context.Context, req *tabletmanagerdatapb.UpdateVReplicationWorkflowsStateRequest) (*tabletmanagerdatapb.UpdateVReplicationWorkflowsStateResponse, error) {
+	if req.GetAllWorkflows() && (len(req.GetIncludeWorkflows()) > 0 || len(req.GetExcludeWorkflows()) > 0) {
+		return nil, vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "cannot specify all workflows along with either of include or exclude workflows")
+	}
```

Comment:
We should move the code that generates the sql query into a separate function. In addition to better code organization it will also be easier to unit test/validate.